snscrape/modules/twitter.py

```diff
@@ -1835,7 +1836,13 @@ def get_items(self):
 						tweetId = int(instruction['entry']['entryId'][6:]) if instruction['entry']['entryId'].startswith('tweet-') else None
 						yield self._graphql_timeline_tweet_item_result_to_tweet(instruction['entry']['content']['itemContent']['tweet_results']['result'], tweetId = tweetId, pinned = True)
 			# Includes tweets by other users on conversations, don't return those
-			for tweet in self._graphql_timeline_instructions_to_tweets(instructions, pinned = False):
+			tweets = list(self._graphql_timeline_instructions_to_tweets(instructions, pinned = False))
+			pageTweetIds = frozenset({tweet.id for tweet in tweets})
+			if len(pageTweetIds) > 0 and pageTweetIds in previousPagesTweetIds:
+				_logger.warning("Found duplicate page of tweets, stopping as assumed cycle found in Twitters pagination")
+				return
```

Comment:
I'd rather see `break` here, although it makes no functional difference currently.